main_viz.py

```diff
@@ -45,28 +39,56 @@ def read_capacity_from_file(path_to_specs):
         return capacity
 
 
-def samples(experiment_name, num_samples=1, batch_size=1, shuffle=True):
-    """ generate a number of samples from the dataset
+def samples(experiment_name, num_samples=1, shuffle=True):
+    """ Generate a number of samples from the dataset.
+
+        Parameters
+        ----------
+        experiment_name : str
+            The name of the experiment, which is the name of the folder that the model is expected to be in.
+
+        num_samples : int
+            The number of samples to load from the dataset
+
+        shuffle : bool
+            Whether or not to shuffle the dataset before drawing samples.
+    """
+    meta_data = load_metadata(os.path.join(RES_DIR, experiment_name))
+    dataset_name = meta_data['dataset']
+
+    data_loader = get_dataloaders(batch_size=num_samples, dataset=dataset_name, shuffle=shuffle)
+
+    data_list = []
+    for batch_idx, (new_data, _) in enumerate(data_loader):
+        if num_samples == batch_idx:
+            break
+        data_list.append(new_data)
+    return torch.cat(data_list, dim=0)
+
+
+def get_sample(experiment_name, samples_to_retrieve):
+    """ Retrieve a particular sample, or set of samples from the dataset.
     """
-    # TODO: use load_metadata in utils.modelIO + don't just use "specs.json"
-    # that might change, so even if load_metadat didn't exist you should import
-    # META_FILENAME. FIlenames should only be defined once in the whole codebase
-    # if not it's hard to maintain!
-    with open('{}/{}/specs.json'.format(RES_DIR, experiment_name)) as spec_file:
-        spec_data = json.load(spec_file)
-        dataset_name = spec_data['dataset']
+    dataset = load_metadata(os.path.join(RES_DIR, experiment_name))["dataset"]
+    if isinstance(samples_to_retrieve, list):
+        num_samples = len(samples_to_retrieve)
+    else:
+        num_samples = 1
+    data_loader = get_dataloaders(dataset, shuffle=False, batch_size=num_samples)
+
+    data_list = []
+    for sample_idx in range(num_samples):
 
-        data_loader = get_dataloaders(batch_size=batch_size, dataset=dataset_name, shuffle=shuffle)
+        if num_samples == 1:
+            return data_loader.dataset[samples_to_retrieve]
```

Comment:
Should this not be before the loop? Not a big deal but then it's not evaluated every iteration.